On GET /ccxcon/... there is an exception regarding hop-by-hop errors

Exception (from @jamiefolsom):

```
web_1   | Traceback (most recent call last):
web_1   |   File "/usr/lib/python2.7/wsgiref/handlers.py", line 85, in run
web_1   |     self.result = application(self.environ, self.start_response)
web_1   |   File "/usr/local/lib/python2.7/dist-packages/django/contrib/staticfiles/handlers.py", line 63, in __call__
web_1   |     return self.application(environ, start_response)
web_1   |   File "/usr/local/lib/python2.7/dist-packages/dj_static.py", line 83, in __call__
web_1   |     return self.application(environ, start_response)
web_1   |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/wsgi.py", line 197, in __call__
web_1   |     start_response(force_str(status), response_headers)
web_1   |   File "/usr/lib/python2.7/wsgiref/handlers.py", line 181, in start_response
web_1   |     assert not is_hop_by_hop(name),"Hop-by-hop headers not allowed"
web_1   | AssertionError: Hop-by-hop headers not allowed
web_1   | [08/Dec/2015 18:57:55] "GET /ccxcon/v1/coursexs/ HTTP/1.1" 500 59
```

--- a/portal/tests/test_ccxcon_api.py
+++ b/portal/tests/test_ccxcon_api.py
@@ -55,6 +55,21 @@ class TestCCXConAPI(TestCase):
         assert resp.content.decode('utf-8') == 'success'
         assert resp['response_header'] == 'response_header_value'
 
+    @requests_mock.mock()
+    def test_hop_by_hop(self, mock):
+        """Test that Keep-Alive and other hop-by-hop headers are filtered out"""
+
+        mock.get(
+            "{base}v1/end/point".format(base=FAKE_CCXCON_API),
+            text="success",
+            headers={"Connection": "Keep-Alive"}
+        )
+
+        resp = self.client.get("{base}v1/end/point".format(base=reverse("ccxcon-api")))
+        assert resp.status_code == 200
+        assert resp.content.decode('utf-8') == 'success'
+        assert 'Connection' not in resp
+
     def test_post(self):
         """
         Test that POST requests are rejected.

--- a/portal/views.py
+++ b/portal/views.py
@@ -5,6 +5,7 @@ Views for teachersportal.
 from __future__ import unicode_literals
 
 import json
+from wsgiref.util import is_hop_by_hop
 
 from django.conf import settings
 from django.shortcuts import render, HttpResponse
@@ -63,7 +64,8 @@ def forward_to_ccxcon(request):
     )
 
     for key, value in response.headers.items():
-        ccxcon_response[key] = value
+        if not is_hop_by_hop(key):
+            ccxcon_response[key] = value
 
     return ccxcon_response
 

